### PR TITLE
isotope support 

### PR DESCRIPTION
**Pull request by vetoshev**
- Added the ability to create isotopic atoms and render .mol ​​for them.
- The choice of isotope is made for atoms if they are present from stable ones.
- Fixed the function for converting text into a superscript because the current implementation did not work correctly for Arial.

### CODE REVIEW

**Comment by BruceAndyLee** [comment]
@vetoshev  don't forget to bump the version in package.json!

We\d like this change reflected on our landing :]

File: src/lib/elements.ts [hunk 1]
```diff
@@ -45,109 +47,109 @@ class ChemicalElement {
...
-    "Lr": new ChemicalElement("Lr", "Lawrencium", 103, 262.1098, [3], 5),
+    "H": new ChemicalElement("H", "Hydrogen", 1, 1.00794, [1], 1, [1,2,3]),
+    "He": new ChemicalElement("He", "Helium", 2, 4.002602, [], 4, [3,4]),
+    "Li": new ChemicalElement("Li", "Lithium", 3, 6.941, [1], 2, [6,7]),
+    "Be": new ChemicalElement("Be", "Beryllium", 4, 9.012182, [2], 4, []),
+    "B": new ChemicalElement("B", "Boron", 5, 10.811, [3], 2, [10,11]),
+    "C": new ChemicalElement("C", "Carbon", 6, 12.011, [4], 1, [12,13,14]),
+    "N": new ChemicalElement("N", "Nitrogen", 7, 14.00674, [3], 1, [14,15]),
+    "O": new ChemicalElement("O", "Oxygen", 8, 15.9994, [2], 1, [16,17,18]),
+    "F": new ChemicalElement("F", "Fluorine", 9, 18.9984032, [1], 1, []),
```
Comment (eizemazal) [inline src/lib/elements.ts:58]: А вот моноизотопные элементы - у них пустой список. Вообще есть радиоактивные изотопы типа 18F, мы не будем их добавлять сюда? Если у тебя пока что  под рукой данные по элементам, может добавим их? Есть некоторые такие изотопы, которые имеют практическое значение (типа 32P, 18F)

File: src/controller/MoleculeEditor.ts [hunk 2]
```diff
@@ -489,6 +493,17 @@ class MoleculeEditor {
         this.menu.visible = true;
     }
 
+    menu_isotopes(vertex: Vertex) {
+        this.menu.clear_buttons();
+        this.menu.add_button( new MenuButton(`x`, `x`, () => { this.commit_action(new ChangeVertexIsotopeAction(this.graph, vertex, 0)); } ));
+        if (vertex.element?.isotopes) {
+            for (const [index, is] of vertex.element.isotopes.entries()) {
+                this.menu.add_button( new MenuButton(`${index + 1}`, `[${is}${vertex.element.symbol}]`, () => { this.commit_action(new ChangeVertexIsotopeAction(this.graph, vertex, is)); } ));
```
Comment (eizemazal) [inline src/controller/MoleculeEditor.ts:501]: а может тут добавим `const iso_idx = int_to_superscript(is);` и будем показывать в меню верхним индексом изотоп?

**Review by eizemazal — CHANGES_REQUESTED** [review]

File: src/graph/Vertex.ts [hunk 3]
```diff
@@ -317,10 +330,10 @@ class Vertex {
         });
         charge_text.setAttr("x", stylesheet.atom_charge_frame_enabled ? hpadding / 2 : 0);
         charge_text.setAttr("y", stylesheet.atom_charge_frame_enabled ? vpadding / 2 : 0);
-        charge_text.setAttr("text", this.charge == -1 ? "－" : this.charge == 1 ? "+" : this.charge > 1 ? `${this._charge}+` : `${Math.abs(this._charge)}-`);
+        charge_text.setAttr("text", this.charge == -1 ? " - " : this.charge == 1 ? "+" : this.charge > 1 ? `${this._charge}+` : `${Math.abs(this._charge)}-`);
```
[the remark this answers is not in the document]
Comment (vetoshev) [reply src/graph/Vertex.ts:333]: Там проблема с символом, который изначально читался - текущий шрифт его не поддерживает
поэтому использовать стал обычный минус
у него проблема что он "липнет" к левой части круга, поэтому обособил пробелами

File: demo/test.html [hunk 4]
```diff
@@ -59,12 +59,13 @@
...
         </script>
         <style type="text/css">
+            @import url('https://fonts.googleapis.com/css2?family=Noto+Sans&display=swap');
             .wrapper {
                 width: 600px;
                 margin: 50px 5px;
             }
             p {
-                font-family: Arial, Helvetica, sans-serif;
+                font-family: "Arial Unicode MS", Helvetica, sans-serif;
```
[the remark this answers is not in the document]
Comment (vetoshev) [reply demo/test.html:68]: Возможно опять проблема с шрифтом - не понимаю
у меня выглядит так:
Снимок экрана 2024-05-30 в 10 56 12

File: src/lib/elements.ts (hunk 1 above)
Comment (vetoshev) [reply src/lib/elements.ts:58]: Я сам поискал какие-то 
Давай добавим еще полезные действительно
Но кажется все мы добавить не сможем(да и не имеет смысла, наверное)

**Comment by vetoshev** [comment]
Не совсем понимаю как писать лицензию 
Посмотрите - https://dejavu-fonts.github.io/

Butlerov Sans — License

The fonts are a modification of the DejaVu and Bitstream Vera fonts (see below). Changes to the DejaVu fonts are in the public domain. The explanation of copyright is provided on the Gnome page on Bitstream Vera fonts. Glyphs imported from the Arev fonts are the property of Tavmjung Bah (see below).

Bitstream Vera Fonts Copyright
Copyright (c) 2003 by Bitstream, Inc. All Rights Reserved. Bitstream Vera is a trademark of Bitstream, Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy of the fonts accompanying this license ("Fonts") and associated documentation files (the "Font Software"), to reproduce and distribute the Font Software, including without limitation the rights to use, copy, merge, publish, distribute, and/or sell copies of the Font Software, and to permit persons to whom the Font Software is furnished to do so, subject to the following conditions:

The above copyright and trademark notices and this permission notice shall be included in all copies of one or more of the Font Software typefaces.

The Font Software may be modified, altered, or added to, and in particular the designs of glyphs or characters in the Fonts may be modified and additional glyphs or characters may be added to the Fonts, only if the fonts are renamed to names not containing either the words "Bitstream" or the word "Vera".

This License becomes null and void to the extent applicable to Fonts or Font Software that has been modified and is distributed under the "Bitstream Vera" names.

The Font Software may be sold as part of a larger software package but no copy of one or more of the Font Software typefaces may be sold by itself.

THE FONT SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO ANY WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT OF COPYRIGHT, PATENT, TRADEMARK, OR OTHER RIGHT. IN NO EVENT SHALL BITSTREAM OR THE GNOME FOUNDATION BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, INCLUDING ANY GENERAL, SPECIAL, INDIRECT, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF THE USE OR INABILITY TO USE THE FONT SOFTWARE OR FROM OTHER DEALINGS IN THE FONT SOFTWARE.

Except as contained in this notice, the names of Gnome, the Gnome Foundation, and Bitstream Inc., shall not be used in advertising or otherwise to promote the sale, use or other dealings in this Font Software without prior written authorization from the Gnome Foundation or Bitstream Inc., respectively. For further information, contact: fonts at gnome dot org.

Arev Fonts Copyright
Original text

Copyright (c) 2006 by Tavmjong Bah. All Rights Reserved.

Permission is hereby granted, free of charge, to any person obtaining a copy of the fonts accompanying this license ("Fonts") and associated documentation files (the "Font Software"), to reproduce and distribute the modifications to the Bitstream Vera Font Software, including without limitation the rights to use, copy, merge, publish, distribute, and/or sell copies of the Font Software, and to permit persons to whom the Font Software is furnished to do so, subject to the following conditions:

The above copyright and trademark notices and this permission notice shall be included in all copies of one or more of the Font Software typefaces.

The Font Software may be modified, altered, or added to, and in particular the designs of glyphs or characters in the Fonts may be modified and additional glyphs or characters may be added to the Fonts, only if the fonts are renamed to names not containing either the words "Tavmjong Bah" or the word "Arev".

This License becomes null and void to the extent applicable to Fonts or Font Software that has been modified and is distributed under the "Tavmjong Bah Arev" names.

The Font Software may be sold as part of a larger software package but no copy of one or more of the Font Software typefaces may be sold by itself.

THE FONT SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO ANY WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT OF COPYRIGHT, PATENT, TRADEMARK, OR OTHER RIGHT. IN NO EVENT SHALL TAVMJONG BAH BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, INCLUDING ANY GENERAL, SPECIAL, INDIRECT, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF THE USE OR INABILITY TO USE THE FONT SOFTWARE OR FROM OTHER DEALINGS IN THE FONT SOFTWARE.

Except as contained in this notice, the name of Tavmjong Bah shall not be used in advertising or otherwise to promote the sale, use or other dealings in this Font Software without prior written authorization from Tavmjong Bah. For further information, contact: tavmjong @ free . fr.

**Review by eizemazal — APPROVED** [review]

**Pull request merged**